server/src/main/java/org/opensearch/gateway/GatewayMetaState.java

```diff
@@ -181,10 +190,14 @@ public void start(
                 } finally {
                     if (success == false) {
                         IOUtils.closeWhileHandlingException(persistedState);
+                        IOUtils.closeWhileHandlingException(remotePersistedState);
                     }
                 }
 
-                this.persistedState.set(persistedState);
+                persistedStateRegistry.addPersistedState(PersistedStateType.LOCAL, persistedState);
```

Comment:
is this flow called when another node in the cluster becomes cluster manager?

Reply:
cluster-manager is static role. So for all cluster manager nodes, it will get initialized (even though that not active leader)